music21/stream/tools.py

```diff
@@ -0,0 +1,201 @@
+        {0.0} <music21.key.KeySignature of 6 sharps>
+        {0.0} <music21.meter.TimeSignature 3/4>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.note.Note C>
+        {2.0} <music21.note.Note D>
+    {3.0} <music21.stream.Measure 2 offset=3.0>
+        {0.0} <music21.clef.TrebleClef>
+        {0.0} <music21.key.KeySignature of 6 sharps>
+        {0.0} <music21.meter.TimeSignature 3/4>
+        {0.0} <music21.note.Note E>
+        {1.0} <music21.note.Note F>
+        {2.0} <music21.note.Note G>
+    {6.0} <music21.stream.Measure 3 offset=6.0>
+        {0.0} <music21.clef.BassClef>
+        {0.0} <music21.key.KeySignature of 5 flats>
+        {0.0} <music21.meter.TimeSignature 2/4>
+        {0.0} <music21.note.Note A>
+        {1.0} <music21.note.Note B>
+    {8.0} <music21.stream.Measure 4 offset=8.0>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.bar.Barline type=final>
+
+    Calling removeDuplicates should remove the duplicates
+    even with those changes now stored within measures,
+    not directly on the part.
+    Specifically, in our example,
+    the duplicates entries are removed from measure 2
+    and the actual changes in measure 3 remain.
+
+    >>> testInPlace = stream.tools.removeDuplicates(p)
+    >>> testInPlace.show('t')
+    {0.0} <music21.stream.Measure 1 offset=0.0>
+        {0.0} <music21.clef.TrebleClef>
+        {0.0} <music21.key.KeySignature of 6 sharps>
+        {0.0} <music21.meter.TimeSignature 3/4>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.note.Note C>
+        {2.0} <music21.note.Note D>
+    {3.0} <music21.stream.Measure 2 offset=3.0>
+        {0.0} <music21.note.Note E>
+        {1.0} <music21.note.Note F>
+        {2.0} <music21.note.Note G>
+    {6.0} <music21.stream.Measure 3 offset=6.0>
+        {0.0} <music21.clef.BassClef>
+        {0.0} <music21.key.KeySignature of 5 flats>
+        {0.0} <music21.meter.TimeSignature 2/4>
+        {0.0} <music21.note.Note A>
+        {1.0} <music21.note.Note B>
+    {8.0} <music21.stream.Measure 4 offset=8.0>
+        {0.0} <music21.note.Note C>
+        {1.0} <music21.bar.Barline type=final>
+
+    As the example shows, this function defaults to working on a stream inPlace.
+
+    >>> testInPlace == p
+    True
+
+    To avoid this, set inPlace to False.
+
+    >>> testNotInPlace = stream.tools.removeDuplicates(p, inPlace=False)
+    >>> testNotInPlace == p
+    False
+
+    This function is primarily designed and intended for use on a stream.Part object.
+    If called on a steam.Score, it will simply be applied to each of that score's parts in turn.
+
+    >>> s = stream.Score()
+    >>> s.append(p)
+    >>> t = stream.tools.removeDuplicates(s, inPlace=False)
+    >>> s.parts[0] == testInPlace
+    True
+
+    '''
+
+    supportedClasses = [meter.TimeSignature, key.KeySignature, clef.Clef]
+
+    removalDict = {}
+
+    if not inPlace:
+        thisStream = deepcopy(thisStream)
+
+    if 'Score' in thisStream.classes:
```

Comment:
Gradually moving away from .classes

```suggestion
    if isinstance(thisStream, stream.Score):
```

And yes, I think this is a fine place to stop instead of worrying about offsets in scores.

Reply:
Ok, noted.